frontend/src/pages/audio/_id/index.vue

```diff
@@ -67,19 +73,29 @@ export default defineComponent({
     const relatedMediaStore = useRelatedMediaStore()
     const searchStore = useSearchStore()
 
-    const { sendCustomEvent } = useAnalytics()
+    const route = useRoute()
+
+    const audio = ref<AudioDetail | null>(singleResultStore.audio)
+
+    useFetch(async () => {
+      const audioId = route.value.params.id
+      await singleResultStore.fetch(AUDIO, audioId)
+
+      const fetchedAudio = singleResultStore.audio
+
+      if (!fetchedAudio) {
+        useContext().error({ statusCode: 404 })
```

Comment:
Is it safe to call `useContext()` inside the `useFetch` callback? Just wondering if it could cause the same ref-based memory leak we had in https://github.com/WordPress/openverse-frontend/issues/1722

Reply:
It's not safe, thank you for catching this! It was a left-over from previous attempts.